### PR TITLE
libcaca: add imlib2, for img previews in ranger

### PR DESCRIPTION
**Pull request by magnetophon**

### CODE REVIEW

File: pkgs/development/libraries/libcaca/default.nix [hunk 1]
```diff
@@ -11,9 +11,9 @@ stdenv.mkDerivation rec {
     sha256 = "1x3j6yfyxl52adgnabycr0n38j9hx2j74la0hz0n8cnh9ry4d2qj";
   };
 
-  configureFlags = "--disable-x11 --disable-imlib2 --disable-doc";
```
Comment (aristidb) [inline pkgs/development/libraries/libcaca/default.nix]: If you add libX11 as a build input, is the --disable-x11 still useful?

Comment (magnetophon) [inline pkgs/development/libraries/libcaca/default.nix]: I don't know why it was there in the first place.
The X11 was wanted by the compiler after I added imlib2.
The pkg also compiles without --disable-x11.

Comment (oxij) [inline pkgs/development/libraries/libcaca/default.nix]: That's a bug in `imglib2`. See https://github.com/oxij/nixpkgs/blob/244fba351c341dfc76efe3f1e7a7711368e36edd/pkgs/applications/networking/browsers/w3m/no-x11.patch for how I fixed this in `w3m`.

**Comment by oxij** [comment]
I think `libcaca` needs `*Support` options. Surely, some deps (and
people) will be unhappy without X11 and some will be unhappy with X11.

**Comment by magnetophon** [comment]
@oxij 
I probably won't be doing that, as [this PR](https://github.com/NixOS/nixpkgs/issues/11222) fixes my issues (img preview in ranger)in the proper way.

**Comment by oxij** [comment]
But having image previews over ssh would be nice too. Not that I'd ever
use it, but seems kinda cool.

**Comment by zimbatm** [comment]
Closing as this PR probably won't be pursued. Feel free to re-open otherwise.

**Pull request closed without merging**